src/nifs/protogalaxy/poly/mod.rs

```diff
@@ -263,4 +401,37 @@ mod test {
             ))
         );
     }
+
+    #[traced_test]
+    #[test]
+    fn zero_g() {
+        let (S, trace) = poseidon_trace();
+        let mut rnd = rand::thread_rng();
+
+        assert!(
+            super::compute_G::<Curve>(Field::random(&mut rnd), &S, trace.clone(), &[trace],)
+                .unwrap()
+                .iter()
+                .all(|f| f.is_zero().into())
+        );
+    }
+
+    #[traced_test]
+    #[test]
+    fn non_zero_g() {
```

Comment:
what is this trying to test?

Reply:
That parallelization works correctly and that no element is missing for initialization

Because `empty_witness` is have only zeroes